CMakeLists.txt

```diff
@@ -240,6 +240,17 @@ cmake_dependent_option(
   "NOT EXECUTORCH_BUILD_ARM_BAREMETAL" OFF
 )
 
+if(EXECUTORCH_BUILD_PYBIND)
```

Comment:
I moved these here to better match the pattern in the rest of the file and also it makes the shared dependencies between training and pybind a little easier to handle

Reply:
This weirdly broke apple builds so undid